tensorflow/python/keras/engine/network_test.py

```diff
@@ -128,6 +128,35 @@ def test_get_updates_bn(self):
     self.assertEqual(len(layer.get_updates_for(x1)), 2)
     self.assertEqual(len(layer.get_updates_for(None)), 0)
 
+  def test_get_layer(self):
+    # create a simple network
+    x = input_layer_lib.Input(shape=(32,))
+    dense_a = layers.Dense(4, name='dense_a')
+    dense_b = layers.Dense(2, name='dense_b')
+    y = dense_b(dense_a(x))
+    network = network_lib.Network(x, y, name='dense_network')
+
+    # test various get_layer by index
+    self.assertEqual(network.get_layer(index=1), dense_a)
+
+    # test invalid get_layer by index
+    with self.assertRaises(ValueError):
```

Comment:
From @fchollet this should check the text of the error with `assertRaisesRegexp`